source/native-plugins/midi-channel-ab.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Carla Native Plugins
+ * Copyright (C) 2012-2015 Filipe Coelho <falktx@falktx.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of
+ * the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * For a full copy of the GNU General Public License see the doc/GPL.txt file.
+ */
+
+#include "CarlaNative.h"
+#include "CarlaMIDI.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+// -----------------------------------------------------------------------
+
+typedef struct {
+    const NativeHostDescriptor* host;
+    bool channels[MAX_MIDI_CHANNELS];
+} MidiGainHandle;
+
+// -----------------------------------------------------------------------
+
+static NativePluginHandle midichanab_instantiate(const NativeHostDescriptor* host)
+{
+    MidiGainHandle* const handle = (MidiGainHandle*)malloc(sizeof(MidiGainHandle));
+
+    if (handle == NULL)
+        return NULL;
+
+    handle->host = host;
+
+    for (int i=MAX_MIDI_CHANNELS; --i>=0;)
+        handle->channels[i] = 0.0f;
```

Comment:
This is not right, the channels are booleans, not floats.
And to zero memory we can use `memset(ptr, 0, full-size)` directly, instead of going through a list.